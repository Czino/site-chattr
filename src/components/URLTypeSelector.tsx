import { atom, useAtom } from 'jotai'
import { keys } from '../helpers/keys'
import { Selector } from './Selector'

type URLType = 'site' | 'page'
export const urlTypeAtom = atom<URLType>('page')
const types: Record<URLType, string> = { page: 'Page', site: 'Site' }
export const URLTypeSelector = () => {
    const [type, setType] = useAtom(urlTypeAtom)

    return (
        <div className="flex flex-row">
            {keys(types).map((key) => (
                <Selector key={key} value={key} onClick={setType} selected={type}>
                    {types[key]}
                </Selector>
            ))}
        </div>
    )
}
