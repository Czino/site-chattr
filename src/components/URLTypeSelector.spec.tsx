import { render, renderHook, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { useAtom } from 'jotai'
import { urlTypeAtom, URLTypeSelector } from './URLTypeSelector'

describe('URLTypeSelector', () => {
    const base = render(<URLTypeSelector />).asFragment()
    it('should render correctly', () => {
        expect(base).toMatchSnapshot()
    })
    it('should select site render correctly', async () => {
        const { result } = renderHook(useAtom, { initialProps: urlTypeAtom })
        expect(result.current[0]).toBe('page')
        const { asFragment } = render(<URLTypeSelector />)
        await userEvent.click(screen.getByText('Site'))
        expect(asFragment()).toMatchDiffSnapshot(base)
        expect(result.current[0]).toBe('site')
    })
})
