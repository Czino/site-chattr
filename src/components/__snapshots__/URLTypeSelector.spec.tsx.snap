// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`URLTypeSelector should render correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-row"
  >
    <button
      class="relative p-4 w-full font-bold text-center uppercase bg-purple-50"
    >
      Page
      <div
        class="absolute right-0 bottom-0 left-0 w-full h-1 bg-highlight"
      />
    </button>
    <button
      class="relative p-4 w-full font-bold text-center uppercase bg-purple-50"
    >
      Site
    </button>
  </div>
</DocumentFragment>
`;

exports[`URLTypeSelector should select site render correctly 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -4,16 +4,16 @@
    >
      <button
        class=\\"relative p-4 w-full font-bold text-center uppercase bg-purple-50\\"
      >
        Page
+       <div
+         class=\\"absolute right-0 bottom-0 left-0 w-full h-1 bg-highlight\\"
+       />
      </button>
      <button
        class=\\"relative p-4 w-full font-bold text-center uppercase bg-purple-50\\"
      >
        Site
-       <div
-         class=\\"absolute right-0 bottom-0 left-0 w-full h-1 bg-highlight\\"
-       />
      </button>
    </div>
  </DocumentFragment>"
`;
